src/operator/setup.rs

```diff
@@ -184,6 +113,103 @@ where
         Ok(())
     }
 
+    /// List all config map belonging to the cluster. Additional labels to filter or limit
+    /// the configmaps via label selector may be passed via `additional_labels`.
+    pub fn list_config_maps(&self, additional_labels: Vec<String>) -> Vec<ConfigMap> {
+        let mut labels = vec![];
+        if let Some(cluster) = &self.cluster {
+            labels.push(format!(
+                "{}={}",
+                stackable_operator::labels::APP_INSTANCE_LABEL,
+                cluster.name(),
+            ));
+        }
+
+        labels.push(format!(
+            "{}={}",
+            stackable_operator::labels::APP_NAME_LABEL,
+            &self.options.cluster_type
+        ));
+
+        labels.extend(additional_labels);
+
+        self.client
+            .list_labeled::<ConfigMap>(&labels.join(","))
+            .items
+    }
+
+    /// Returns all available nodes in the cluster. Can be used to determine the expected pods
+    /// for tests (depending on the custom resource)
+    pub fn list_nodes(&self) -> Vec<Node> {
+        self.client
+            .list_labeled::<Node>("kubernetes.io/arch=stackable-linux")
+            .items
+    }
+
+    /// List all available pods in the cluster via the name label.
+    pub fn list_pods(&self) -> Vec<Pod> {
+        let mut labels = vec![];
+        if let Some(cluster) = &self.cluster {
+            labels.push(format!(
+                "{}={}",
+                stackable_operator::labels::APP_INSTANCE_LABEL,
+                cluster.name(),
+            ));
+        }
+
+        labels.push(format!(
+            "{}={}",
+            stackable_operator::labels::APP_NAME_LABEL,
+            &self.options.cluster_type
+        ));
+
+        self.client.list_labeled::<Pod>(&labels.join(",")).items
+    }
+
+    /// Write a formatted message with cluster kind and cluster name in the beginning to the console.
+    fn log(&self, message: &str) -> String {
+        let cluster_name = if self.cluster.is_some() {
+            T::name(self.cluster.as_ref().unwrap())
+        } else {
+            "<not-found>".to_string()
+        };
+        format!("[{}/{}] {}", T::kind(&()), cluster_name, message)
+    }
+
+    /// This creates a kube client and should be executed at the start of every test.
+    pub fn new(options: TestClusterOptions, timeouts: TestClusterTimeouts) -> Self {
```

Comment:
nit: I would prefer this to be at the top of the impl block.

Reply:
done